runtime/lua/vim/lsp/_tagfunc.lua

```diff
@@ -47,18 +92,21 @@ local function query_definition(pattern)
       end
     end
   end
+
   return results
 end
 
 ---@param pattern string
----@return table[]
+---@return {name: string, filename: string, cmd: string, kind?: string}[]
```

Comment:
Might be a good idea to create an alias for this object type since you repeat it in several places.